docs/source/release_notes.rst

```diff
@@ -6,6 +6,7 @@ Release Notes
         * Allowed the DFS Transformer to calculate feature values for Features with a ``dataframe_name`` that is not ``"X"`` :pr:`3873`
     * Changes
         * Reduced the threshold for setting ``use_covariates`` to False for ARIMA models in AutoMLSearch :pr:`3868`
+        * Changed ``BooleanNullable`` inference logic :pr:`3858`
```

Comment:
Can you add a line about upgrading the woodwork version as well? (only really useful since there's the note about pinning it in here)